### ISSUE TITLE
Where's regex, host, scheme, catch-all matching ? 

### ISSUE THREAD

**Issue by Thinkscape** [DASPRiD/Dash#66]
https://github.com/zendframework/zend-mvc/tree/master/src/Router/Http

Just hoping you have that thought out, as it might screw up your current architecture.
There's tons of use-cases where those route types are required.

**Comment by DASPRiD** [comment]
There is no catch-all route supplied by default (although in the new architecture, catch-all would be implemented as a path parser, not as an entire route). Hostname and scheme matching are handled by the generic route, see the README. And so far nobody could show me a valid example anymore where the segment parser couldn't handle anything a pure regex parser could.

**Comment by Thinkscape** [comment]
The syntax is somewhat simpler, but confusing :-\ 
I've now searched through readme, and yeah, I see there are a few "hidden" keys you can use to get some behavior... in addition to the [0], [1], [2] keys.

**Comment by Thinkscape** [comment]
> valid example anymore where the segment parser couldn't handle anything a pure regex parser could

really ? I use regex routes all the time ...

```
'route' => '#download-file/id-(?<id>0-9)+/(?<filename>.*?)#'

**Comment by DASPRiD** [comment]
Doesn't seem like you need a regex at all:
```'download-file/id-:id/:filename'``` (with the constraints defined properly).

Although I'm thinking about integrating the constraints directly into the route pattern like some other routers do.

**Comment by Thinkscape** [comment]
Nope, because your route does not do catchall of the filename... and there's no way to do backtracking, alternatives, lookahead/behinds etc. Also - can't control greediness or utf8 handling, or case sensitivity ...

**Comment by DASPRiD** [comment]
Sure it can do catch-all of the filename, simply define it's constraint to be ```.*?```. The constraint can also contain alternatives and such. And again, the rare cases where a segment route can't handle something, you can always use another path parser.

**Comment by Thinkscape** [comment]
> you can always use another path parser
Regex parser? :dash:

**Comment by DASPRiD** [comment]
Yes, for instance. Although I'm not supplying one by default.

**Comment by Thinkscape** [comment]
Why not? That's another step back :\
